rafx-framework/src/render_features/jobs/extract/render_feature_extract_job.rs

```diff
@@ -0,0 +1,43 @@
+use crate::render_features::render_features_prelude::*;
+use std::ops::Range;
+
+/// A type-erased trait used by the `Renderer`, `RenderFrameJob`, and `RendererThreadPool`
+/// to control the workload of the rendering process without identifying specific types
+/// used in each `RenderFeature`'s frame packet or workload. See `ExtractJob` and the
+/// `ExtractJobEntryPoints` for implementation details.
+pub trait RenderFeatureExtractJob<'extract>: Send + Sync {
+    fn begin_per_frame_extract(&self);
+
+    fn extract_render_object_instance(
+        &self,
+        range: Range<usize>,
+    );
+
+    fn view_packet(
+        &self,
+        view_index: ViewFrameIndex,
+    ) -> &dyn RenderFeatureViewPacket;
+
+    fn extract_render_object_instance_per_view(
+        &self,
+        view_packet: &dyn RenderFeatureViewPacket,
+        range: Range<usize>,
+    );
+
+    fn end_per_view_extract(
+        &self,
+        view_packet: &dyn RenderFeatureViewPacket,
+    );
+
+    fn end_per_frame_extract(&self);
```

Comment:
is there documentation for these elsewhere? if not, I'm sure it's something you already planned to add :)

Reply:
Yes, specifically in `ExtractJobEntryPoints` (and `RenderFeatureExtractJob` does have a comment that points users to look at the entry points instead, ditto for `ExtractJob`)